libretroshare/src/pqi/pqissllistener.cc

```diff
@@ -830,91 +763,73 @@ int pqissllistener::status()
 }
 
 int pqissllistener::completeConnection(int fd, IncomingSSLInfo& info)
-{ 
+{
+	constexpr int failure = -1;
+	constexpr int success = 1;
 
 	// Get the Peer Certificate....
-	X509 *peercert = SSL_get_peer_certificate(info.ssl);
-
-	if (peercert == NULL)
+	X509* peercert = SSL_get_peer_certificate(info.ssl);
+	if(!peercert)
 	{
-  	        pqioutput(PQL_WARNING, pqissllistenzone, 
-		 "pqissllistener::completeConnection() Peer Did Not Provide Cert!");
-
-		// failure -1, pending 0, sucess 1.
-		// pqissllistenbase will shutdown!
-		return -1;
+		RsFatal() << __PRETTY_FUNCTION__ << " failed to retrieve peer "
+		          << "certificate at this point this should never happen!"
+		          << std::endl;
+		print_stacktrace();
+		exit(failure);
 	}
 
-	// Check cert.
-	RsPeerId newPeerId;
-
+	RsPgpId pgpId = RsX509Cert::getCertIssuer(*peercert);
+	RsPeerId newPeerId = RsX509Cert::getCertSslId(*peercert);
 
-	/****
-	 * As the validation is actually done before this...
-	 * we should only need to call CheckCertificate here!
-	 ****/
+#ifdef RS_PQISSL_AUTH_REDUNDANT_CHECK
```

Comment:
you should remove that ifdef too. These checks are here for double-check security reasons. Your comment below will tell that the check is redundant, but it is still important to do it.

Reply:
RS_PQISSL_AUTH_REDUNDANT_CHECK = 1,  the ifdef is there only to make it easy to test that those checks are redundant, but enabling/disabling it in a testing session and verifying the behaviour is the same without changing the code

Reply:
RS_PQISSL_AUTH_REDUNDANT_CHECK = 1,  the ifdef is there only to make it easy to test that those checks are redundant, but enabling/disabling it in a testing session and verifying the behaviour is the same without changing the code

Reply:
Renamed to RS_PQISSL_AUTH_DOUBLE_CHECK